src/Message.cpp

```diff
@@ -583,19 +583,37 @@ void Message::rewind(bool complete)
 
 std::string Message::getInterfaceName() const
 {
-    return sd_bus_message_get_interface((sd_bus_message*)msg_);
+    auto interface = sd_bus_message_get_interface((sd_bus_message*)msg_);
+    SDBUS_THROW_ERROR_IF(!interface, "Failed to get message interface", EINVAL);
+    return interface;
```

Comment:
If we go for empty strings for nonexisting sd-bus message data, we could have this here (and in other methods in this PR):

```suggestion
    auto interface = sd_bus_message_get_interface((sd_bus_message*)msg_);
    return interface != nullptr ? {interface} : {};
```

Reply:
Nice, but unfortunately does not work for me (gcc version 10.3.1):
```
sdbus-cpp/src/Message.cpp: In member function ‘std::string sdbus::Message::getInterfaceName() const’:
sdbus-cpp/src/Message.cpp:587:35: error: expected primary-expression before ‘{’ token 
  587 |     return interface != nullptr ? {interface} : {};
      |                                   ^
sdbus-cpp/src/Message.cpp:587:34: error: expected ‘:’ before ‘{’ token
  587 |     return interface != nullptr ? {interface} : {};
      |                                  ^~
      |                                  :
sdbus-cpp/src/Message.cpp:587:35: error: expected primary-expression before ‘{’ token 
  587 |     return interface != nullptr ? {interface} : {};
```

Reply:
OK, I see, seems not to work with ternary operator. Your approach is fine. Thanks